src/agent/src/unix_daemon.cpp

```diff
@@ -91,51 +89,14 @@ namespace unix_daemon
         return true;
     }
 
-    std::string LockFileHandler::GetExecutablePath()
-    {
-        std::vector<char> pathBuffer(PATH_MAX);
-
-#if defined(__linux__)
-        ssize_t len = readlink("/proc/self/exe", pathBuffer.data(), pathBuffer.size() - 1);
-        if (len != -1)
-        {
-            pathBuffer[static_cast<std::size_t>(len)] = '\0';
-            return std::filesystem::path(pathBuffer.data()).parent_path().string();
-        }
-        else
-        {
-            throw std::runtime_error("Failed to retrieve executable path on Linux");
-        }
-
-#elif defined(__APPLE__)
-        uint32_t size = static_cast<uint32_t>(pathBuffer.size());
-        if (_NSGetExecutablePath(pathBuffer.data(), &size) == 0)
-        {
-            return std::filesystem::path(pathBuffer.data()).parent_path().string();
-        }
-        else
-        {
-            pathBuffer.resize(size);
-            if (_NSGetExecutablePath(pathBuffer.data(), &size) == 0)
-            {
-                return std::filesystem::path(pathBuffer.data()).parent_path().string();
-            }
-            else
-            {
-                throw std::runtime_error("Failed to retrieve executable path on macOS");
-            }
-        }
-#endif
-    }
-
-    LockFileHandler GenerateLockFile()
+    LockFileHandler GenerateLockFile(std::string lockFilePath)
     {
-        return {};
+        return {LockFileHandler(std::move(lockFilePath))};
     }
 
-    std::string GetDaemonStatus()
+    std::string GetDaemonStatus(std::string lockFilePath)
     {
-        LockFileHandler lockFileHandler = GenerateLockFile();
+        LockFileHandler lockFileHandler = GenerateLockFile(std::move(lockFilePath));
```

Comment:
Shouldn´t LockFileHandler remove the LockFile on destruction?

Reply:
I did not add the file deletion in the destructor of the function because when a new lock is created or its status is checked, instances of this object are also created, causing issues when attempting to delete the file. Instead, I directly delete the file when the agent stops running properly in [here](https://github.com/wazuh/wazuh-agent/blob/4923ec1840be0c2c82901168ffac8f2a306ac6f3/src/agent/src/process_options_unix.cpp#L30.) . What do you think?

Reply:
So what happens if the agent crashes and the lock file is not removed? Sorry I know the question may not be 100% related to this PR, but I lost track of this part.

Reply:
I'm asking because the original idea, before lockfiles (when this was PIDFIleHandler), was to delete the file on shutdown, RAII style. Status checking function was static to avoid creating an object. I don´t know if that still makes sense, but if it does, it's always better. I'm fine either way.

Reply:
@jr0me If the agent crashes, the lock on the file disappears, allowing a new agent instance to lock the same file again.

Regarding where the file is deleted, @aritosteles, the issue arises because a new agent instance could open a file descriptor for the file but fail to acquire the lock. It would then close the file, causing the LockFileHandler instance to disappear, resulting in an error when attempting to delete a file it does not own, or potentially deleting it (I should test this particular scenario).